### DISCUSSION TITLE
Implementing Integral Constraints in DeepXDE

### DISCUSSION THREAD

**Discussion by ppandit95** [lululxvi/deepxde#1749] [General]
Dear Dr @lululxvi 

I am trying to fit the composition profiles in a ternary alloy diffusion process obtained from EPMA which satisfy a physical constraint in integral form (i.e. the point where area under the composition curve and the area above the composition curve should remain the same for all elements in ternary alloy).In this regard, any resources would be beneficial.

Many Thanks
Pushkar

**Comment by yogibadepalli** [comment]
Dear @ppandit95,

I'm currently working on implementing an integral constraint to normalize the output of a neural network, specifically using the following condition:

(\int_{0}^{\infty} |r^2| \, dr = 1)

To achieve this, I need to incorporate the integral constraint into the training process (**deepxde**). I noticed from your earlier query that you were exploring a similar approach. I wanted to reach out and ask if you’ve found a solution or workaround for implementing this constraint effectively.

Any insights or suggestions you can share would be greatly appreciated.

Best Regards,
Yogeswara.

**Comment by lululxvi** [comment]
Check the DeepXDE paper and Integro-differential equations https://deepxde.readthedocs.io/en/latest/demos/pinn_forward.html
